src/utils/PokeApi.jsx

```diff
@@ -0,0 +1,62 @@
+export const baseUrl = "https://pokeapi.co/api/v2/pokemon";
+
+export async function fetchPokemon() {
+  function arrangeTypes(data) {
+    const typeHolder = document.getElementById("pokeType");
+    const pokeTypes = data.types;
+    if (pokeTypes[1]) {
+      typeHolder.textContent =
+        pokeTypes[0].type.name + " " + pokeTypes[1].type.name;
+    } else {
+      return (typeHolder.innerText = pokeTypes[0].type.name);
+    }
+  }
+
+  try {
+    const pokemonName = document
+      .getElementById("pokemonName")
+      .value.toLowerCase();
+    const response = await fetch(`${baseUrl}/${pokemonName}`);
+
+    if (!response.ok) {
+      throw new Error("Could not find resource");
+    }
+
+    const data = await response.json();
+    console.log(data);
+
+    const pokeSprite = data.sprites.front_default;
+    const pokeName = data.name;
+    const pokeType = arrangeTypes(data);
+    const pokeId = data.id;
+    const pokeHp = data.stats[0].base_stat;
+    const pokeAtk = data.stats[1].base_stat;
+    const pokeDef = data.stats[2].base_stat;
+    const pokeSpatk = data.stats[3].base_stat;
+    const pokeSpdef = data.stats[4].base_stat;
+    const pokeSpd = data.stats[5].base_stat;
+
+    const imgElement = document.getElementById("pokeSprite");
+    const nameElement = document.getElementById("pokeName");
+    const idElement = document.getElementById("pokeId");
+    const hpElement = document.getElementById("pokeHp");
+    const atkElement = document.getElementById("pokeAtk");
+    const defElement = document.getElementById("pokeDef");
+    const spatkElement = document.getElementById("pokeSpatk");
+    const spdefElement = document.getElementById("pokeSpdef");
+    const spdElement = document.getElementById("pokeSpd");
```

Comment:
`document.querySelector` or `getElementById` should **not be used** in `React` because it’s a **very slow** operation for the browser. React was specially created not to search anything in the DOM. So, you can directly operate with the markup inside the components. Or you can use hook `useRef` for the reference.


You should **retrun** the fetched data from the function and use it in your component to render the details